chain/client/src/tx_request_handler.rs

```diff
@@ -0,0 +1,420 @@
+use near_async::actix_wrapper::SyncActixWrapper;
+use near_async::messaging::CanSend;
+use near_async::messaging::Handler;
+use near_async::time::Clock;
+use near_chain::Chain;
+use near_chain::ChainGenesis;
+use near_chain::types::RuntimeAdapter;
+use near_chain::types::Tip;
+use near_chain_configs::ClientConfig;
+use near_chain_configs::MutableValidatorSigner;
+use near_chunks::client::ShardedTransactionPool;
+use near_epoch_manager::EpochManagerAdapter;
+use near_epoch_manager::shard_assignment::account_id_to_shard_id;
+use near_epoch_manager::shard_assignment::shard_id_to_uid;
+use near_epoch_manager::shard_tracker::ShardTracker;
+use near_network::client::ProcessTxRequest;
+use near_network::client::ProcessTxResponse;
+use near_network::types::NetworkRequests;
+use near_network::types::PeerManagerAdapter;
+use near_network::types::PeerManagerMessageRequest;
+use near_pool::InsertTransactionResult;
+use near_primitives::stateless_validation::ChunkProductionKey;
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::EpochId;
+use near_primitives::types::ShardId;
+use near_primitives::unwrap_or_return;
+use near_primitives::validator_signer::ValidatorSigner;
+use std::collections::HashSet;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use crate::metrics;
+
+pub type TxRequestHandlerActor = SyncActixWrapper<TxRequestHandler>;
+
+impl Handler<ProcessTxRequest> for TxRequestHandler {
+    fn handle(&mut self, msg: ProcessTxRequest) -> ProcessTxResponse {
```

Comment:
Hm, so what's suspicious to me is this taking a `&mut self` even though this is supposed to run on multiple threads. `&mut self` must not alias after all, so the only way to achieve parallelism otherwise would be to have multiple `Self`s, one for each thread that can process the actor… But `TxRequestHandler` is not `Clone` so how does this actually achieve multi-threading?